### PR TITLE
Load Solidus engine extension files automatically

### PR DESCRIPTION
**Pull request by aldesantis**
Improves `EngineExtensions::Decorators` to automatically load the right files when Solidus engines are installed. For instance, if solidus_backend is installed, it will tell Rails to load `lib/views/backend` and `lib/controllers/backend`, as well as the decorators in `lib/decorators/backend`.

This also renames `SolidusSupport::EngineExtensions::Decorators` to `SolidusSupport::EngineExtensions`, since the module doesn't only deal with decorators anymore.

### CODE REVIEW

File: lib/solidus_support/engine_extensions/decorators.rb [hunk 1]
```diff
@@ -4,33 +4,62 @@ module SolidusSupport
...
-          base_path = root.join('app/decorators')
-
           if Rails.respond_to?(:autoloaders) && Rails.autoloaders.main
-            # Add decorators folder to the Rails autoloader. This
-            # allows Zeitwerk to resolve decorators paths correctly,
-            # when used.
-            Dir.glob(base_path.join('*')) do |decorators_folder|
+            # Add decorators folder to the Rails autoloader. This tells Zeitwerk to treat paths
+            # such as app/decorators/controllers as roots.
+            Dir.glob(decorators_root.join('*')) do |decorators_folder|
```
Comment (kennyadsl) [inline lib/solidus_support/engine_extensions/decorators.rb]: `decorators_root.glob('*')`

File: lib/solidus_support/engine_extensions/decorators.rb [hunk 2]
```diff
@@ -4,33 +4,62 @@ module SolidusSupport
...
+
+        private
+
+        # Loads decorator files.
+        #
+        # This is needed since they are never explicitly referenced in the application code and
+        # won't be loaded by default. We need them to be executed regardless in order to decorate
+        # existing classes.
+        def load_decorators_from(path)
+          Dir.glob(path.join('**/*.rb')) do |decorator_path|
```
Comment (kennyadsl) [inline lib/solidus_support/engine_extensions/decorators.rb]: `path.glob('**/*.rb')`

**Review by kennyadsl — COMMENTED** [review]
Super! Just left a couple of things.

**Comment by spaghetticode** [comment]
@aldesantis I have one concern. 

We're adding new private methods to a class that inherits form `Rails::Engine`, a class that we don't own. I wonder if the new methods, having a rather generic name, may end up in overriding the ones that could be defined by the superclass in a remote but still plausible future. 

So, instead of for example `decorators_root`, would something like `solidus_decorators_root` make more sense?

**Comment by aldesantis** [comment]
@spaghetticode I think that's a good point. It seems unlikely that Rails will officially introduce the concept of engines at some point, but it doesn't hurt to give the methods names that are a bit more specific.

**Review by spaghetticode — APPROVED** [review]
@aldesantis thank you 👍

**Review by kennyadsl — APPROVED** [review]
C O O L!

**Pull request merged**